### PR TITLE
Kafka

### PR DESCRIPTION
**Pull request by BerniVarga**
Features:
* App structure for initial Kafka setup
* App structure for initial Zookeeper setup
* Breakdown of promptu functionality for kafka: producer, consumer and feeder (currently still reading from the DB)

TODOs (can be addressed later, separately):
* terraform config for all the above apps (+ ip v4 for kafka, zookeeper + volume for kafka + machine setup - raised kafka memory to 2GB)
* creating multiple partitions/topic
* creating multiple brokers for kafka and zookeeper
* change feeder to use kafka topics + DB (only at restart)
* possibly looking into a UI to looking into kafka instead of using CMD [not priority]

### CODE REVIEW

File: apps/post-producer/main.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
+package main
```
Comment (sansaid) [inline apps/post-producer/main.go:1]: Naming suggestion for service: `post-api` - just to make it clear we split the former `api` directory

File: apps/post-feeder/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+FROM golang:1.18 as base
```
Comment (sansaid) [inline apps/post-feeder/Dockerfile:1]: Naming suggestion for service: `feeder-api` - just to make it clear we split the former `api` directory

File: apps/post-producer/main.go (hunk 1 above)
Comment (BerniVarga) [reply apps/post-producer/main.go:1]: Will do in a separate PR.

File: apps/post-feeder/Dockerfile (hunk 2 above)
Comment (BerniVarga) [reply apps/post-feeder/Dockerfile:1]: Will do in a separate PR.

File: apps/post-db-updater/main.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
+package main
```
[the remark this answers is not in the document]
Comment (BerniVarga) [reply apps/post-db-updater/main.go]: Hm, with that name I feel like we would be missing the actual responsibility of the service from the name (saving new messages). It is indeed a consumer, but I feel like that can be an implementation detail.

I can't really think of a name that sounds good with both consumer and db, but i'm happy to change it if we can come up with something.

**Review by sansaid — APPROVED** [review]

**Pull request merged**